icu4j/main/classes/core/src/com/ibm/icu/impl/Punycode.java

```diff
@@ -263,6 +269,11 @@ private static boolean isSurrogate(int ch){
     public static StringBuilder decode(CharSequence src, boolean[] caseFlags)
                                throws StringPrepParseException{
         int srcLength = src.length();
+        if (srcLength > 2000) {
```

Comment:
Define this 2000 as a name private static w/ a meaningful name please.

Reply:
ditto